staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -888,8 +889,15 @@ func (c *Cacher) GetList(ctx context.Context, key string, opts storage.ListOptio
 
 	objs, readResourceVersion, indexUsed, err := c.listItems(ctx, listRV, preparedKey, pred, recursive)
 	if err != nil {
+		if consistentRead && storage.IsTooLargeResourceVersion(err) {
+			metrics.RecordConsistentRead(c.resourcePrefix, false)
+			return c.storage.GetList(ctx, key, opts, listObj)
```

Comment:
How can we know what percentage of these calls succeeded?

Reply:
`apiserver_watch_cache_consistent_read_total{success="true"}/(apiserver_watch_cache_consistent_read_total{success="true"}+apiserver_watch_cache_consistent_read_total{success="false"})

Reply:
Please see https://github.com/kubernetes/community/pull/7708/files

Reply:
> apiserver_watch_cache_consistent_read_total{success="true"}/(apiserver_watch_cache_consistent_read_total{success="true"}+apiserver_watch_cache_consistent_read_total{success="false"})

Intuitively, this appears to be the percent of consistent reads that result calling line 894.  But I don't see how this discriminates between this list succeeding and failing.  Perhaps walking through examples will help.


1. consistent read fails on something other than istoolarge: I think this is not included in `ConsistentReadTotal` at all.
2. consistent read success: I think this is `ConsistentReadTotal` labeled with `success=true`
3. consistent read fail on istoolarge, backup list succeeds: I think this is `ConsistentReadTotal` labeled with `success=false`
4. consistent read fail on istoolarge, backup list fails: I think this is `ConsistentReadTotal` labeled with `success=false`

Answers I'd like to be able to get from metrics:
1. there have been four total consistent read requests.  I think currently I will think there are three, not four.
2. 25% of consistent reads succeed.  I think currently I will believe that is 33%.
3. 50% of backup lists have succeeded.  I think currently there is no way to determine.